### PR TITLE
EA-114: 3 out of 4 getUniqueDiagnoses(..) cases re-implemented with HQL.

### PR DESCRIPTION
**Pull request by mks-d**

### CODE REVIEW

**Comment by mks-d** [comment]
@dkayiwa, if you're ok with all this, could you create a new 1.12.x branch too so that I can PR the same change out there?

File: api/src/test/java/org/openmrs/module/emrapi/visit/VisitDomainWrapperComponentTest.java [hunk 1]
```diff
@@ -151,5 +171,171 @@ public void getVisitTemplate_shouldReturnNullIfNoMatchingVisitAttributes() {
...
+          break;
+       case LEGACY:
+          res = visitDomainWrapper.getUniqueDiagnosesLegacy(primaryOnly, confirmedOnly);
+          whichOne = impl.name();
+          break;
+       default:
+          Assert.fail("Invalid implementation specified.");
+       }
+       long time1 = System.currentTimeMillis();
+       System.out.println(whichOne + " getUniqueDiagnoses(..) | primaryOnly=" + primaryOnly.toString() + ", confirmedOnly=" + confirmedOnly.toString());
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/module/emrapi/visit/VisitDomainWrapperComponentTest.java]: Do you still need these println statements?

**Comment by dkayiwa** [comment]
@mks-d i seem to have forgotten. Could you remind me about the reason for creating the 1.12.x branch? :)

File: api/src/test/java/org/openmrs/module/emrapi/visit/VisitDomainWrapperComponentTest.java (hunk 1 above)
Comment (mks-d) [inline api/src/test/java/org/openmrs/module/emrapi/visit/VisitDomainWrapperComponentTest.java]: I can just comment them out if you don't want the build output log to be affected, would that be ok?

**Comment by mks-d** [comment]
Same reason as for the existence of the Core Apps 1.7.x branch. This is to backport changes in the Ref App 2.3.x.

**Comment by mks-d** [comment]
Thanks @dkayiwa!

**Comment by dkayiwa** [comment]
@mks-d here it is https://github.com/openmrs/openmrs-module-emrapi/tree/1.12.x

**Comment by mks-d** [comment]
@dkayiwa, thanks but could I ask you to redo it setting the POMs to version 1.12.1-SNAPSHOT (instead of 1.12-SNAPSHOT)?
Otherwise it would suggest that we are heading to release 1.12 again.

**Comment by dkayiwa** [comment]
@mks-d how about if you just do a PR and change the pom files to 1.12.1-SNAPSHOT?

**Comment by mks-d** [comment]
Done here: https://github.com/openmrs/openmrs-module-emrapi/pull/111

**Comment by dkayiwa** [comment]
@mks-d that seems to have merge conflicts.

**Pull request merged**